core/project_settings.cpp

```diff
@@ -368,34 +368,32 @@ Error ProjectSettings::_setup(const String &p_path, const String &p_main_pack, b
 		String exec_filename = exec_path.get_file();
 		String exec_basename = exec_filename.get_basename();
 
-		// Try to load data pack at the location of the executable
-		// As mentioned above, we have two potential names to attempt
-
-		if (_load_resource_pack(exec_dir.plus_file(exec_basename + ".pck")) ||
-				_load_resource_pack(exec_dir.plus_file(exec_filename + ".pck"))) {
+		// Attempt with PCK bundled into executable
+		if (_load_resource_pack(exec_path)) {
 			found = true;
-		} else {
-			// If we couldn't find them next to the executable, we attempt
-			// the current working directory. Same story, two tests.
-			if (_load_resource_pack(exec_basename + ".pck") ||
-					_load_resource_pack(exec_filename + ".pck")) {
-				found = true;
-			}
 		}
 
-#ifdef OSX_ENABLED
-		// Attempt to load PCK from macOS .app bundle resources
 		if (!found) {
+#ifdef OSX_ENABLED
+			// Attempt to load PCK from macOS .app bundle resources
 			if (_load_resource_pack(OS::get_singleton()->get_bundle_resource_dir().plus_file(exec_basename + ".pck"))) {
 				found = true;
 			}
-		}
 #endif
 
-		// Attempt with PCK bundled into executable
-		if (!found) {
-			if (_load_resource_pack(exec_path)) {
+			// Try to load data pack at the location of the executable
+			// As mentioned above, we have two potential names to attempt
+
+			if (_load_resource_pack(exec_dir.plus_file(exec_basename + ".pck")) ||
```

Comment:
And this part probably should be wrapped in `if (!found)` or it will load pack from the exec_dir on top of embedded.

Reply:
That is in the `!found` block. I moved the `OSX_ENABLED` block into the `!found` block and removed `! found` inside the `ifdef`

Reply:
Oh my mistake. wait

Reply:
Each one should be it the separate (!found) block to run if the previous one failed.

Reply:
done. I oversaw that, when I moved the `ifdef`

Reply:
in theory we could get rid of the `if (... || ...) found = true`and set it directly, but I don't know if'd break style conventions there.